src/config.js

```diff
@@ -327,13 +356,63 @@ export function newConfig() {
       .forEach(listener => listener.callback(options));
   }
 
+  function setBidderConfig(config) {
+    try {
+      check(config);
+      Object.keys(config).forEach(bidder => {
+        check(config[bidder]);
+        if (!bidderConfig[bidder]) {
+          bidderConfig[bidder] = {}
+        }
+        Object.keys(config[bidder]).forEach(topic => {
+          let option = config[bidder][topic];
+          if (utils.isPlainObject(option)) {
+            bidderConfig[bidder][topic] = Object.assign({}, bidderConfig[bidder][topic] || {}, option);
+          } else {
+            bidderConfig[bidder][topic] = option;
+          }
+        });
+      });
+    } catch (e) {
+      utils.logError(e);
+    }
+    function check(obj) {
+      if (!utils.isPlainObject(obj)) {
+        throw 'setBidderConfig bidder options must be an object';
+      }
+    }
+  }
+
+  /**
+   * Internal functions for core to execute some synchronous code while having an active bidder set.
+   */
+  function runWithBidder(bidder, fn) {
+    currBidder = bidder;
+    try {
+      return fn();
+    } finally {
+      currBidder = null;
+    }
+  }
+  function callbackWithBidder(bidder) {
+    return function(cb) {
+      return function(...args) {
+        return runWithBidder(bidder, utils.bind.call(cb, this, ...args))
+      }
+    }
+  }
+
   resetConfig();
 
   return {
     getConfig,
     setConfig,
     setDefaults,
-    resetConfig
+    resetConfig,
+    runWithBidder,
+    callbackWithBidder,
+    setBidderConfig,
+    getBidderConfig
```

Comment:
we just have to make sure no adapter accesses this config method correct?

Reply:
yes. they'd have to import it directly so it'd be pretty obvious. maybe could add a linter check that bid adapters can't import this method.